norfair/drawing/draw_boxes.py

```diff
@@ -0,0 +1,178 @@
+        Thickness or width of the line.
+    draw_labels : bool, optional
+        If set to True, the label is added to a title that is drawn on top of the box.
+        If an object doesn't have a label this parameter is ignored.
+    draw_ids : bool, optional
+        If set to True, the id is added to a title that is drawn on top of the box.
+        If an object doesn't have an id this parameter is ignored.
+    text_size : Optional[float], optional
+        Size of the title, the value is used as a multiplier of the base size of the font.
+        By default the size is scaled automatically based on the frame size.
+    text_color : Optional[int], optional
+        Color of the text. By default the same color as the box is used.
+    text_thickness : Optional[int], optional
+        Thickness of the font. By default it's scaled with the `text_size`.
+
+    Returns
+    -------
+    np.ndarray
+        The resulting frame.
+    """
+    #
+    # handle deprecated parameters
+    #
+    if random_color is not None:
+        warn_once(
+            'Parameter "random_color" is deprecated, set `color="random"` instead'
+        )
+        color = "random"
+    if color_by_label is not None:
+        warn_once(
+            'Parameter "color_by_label" is deprecated, set `color="by_label"` instead'
+        )
+        color = "by_label"
+    if detections is not None:
+        warn_once('Parameter "detections" is deprecated, use "drawables" instead')
+        drawables = detections
+    if line_color is not None:
+        warn_once('Parameter "line_color" is deprecated, use "color" instead')
+        color = line_color
+    if line_width is not None:
+        warn_once('Parameter "line_width" is deprecated, use "thickness" instead')
+        thickness = line_width
+    if label_size is not None:
+        warn_once('Parameter "label_size" is deprecated, use "text_size" instead')
+        text_size = label_size
+    # end
+
+    if color is None:
+        color = "by_id"
+    if thickness is None:
+        thickness = int(max(frame.shape) / 500)
+
+    if drawables is None:
+        return frame
+
+    for obj in drawables:
+        d = Drawable(obj)
+
+        if color == "by_id":
+            obj_color = Palette.choose_color(d.id)
+        elif color == "by_label":
+            obj_color = Palette.choose_color(d.label)
+        elif color == "random":
+            obj_color = Palette.choose_color(np.random.rand())
+        else:
+            obj_color = parse_color(color)
+
+        points = d.points.astype(int)
+        if draw_box:
+            Drawer.rectangle(
+                frame,
+                tuple(points),
+                color=obj_color,
+                thickness=thickness,
+            )
+
+        text = _build_text(d, draw_labels=draw_labels, draw_ids=draw_ids)
+        if text:
+            if text_color is None:
+                obj_text_color = obj_color
+            else:
+                obj_text_color = color
```

Comment:
Shouldn't we assign `text_color` or its parsed color to `obj_text_color` instead of `color`?

Reply:
Good catch! I need to parse `text_color` and use that value here